src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +201,37 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        if !cfg!(target_os = "linux") {
```

Comment:
I’m not necessarily opposed to having default on linux only, but did you consider the fact that FreeBSD, OpenBSD, DragonflyBSD, and similar UNIX-like OSes use ELF and are likely to have working ld.gold as well?